search-processors/src/main/java/org/opensearch/searchpipelines/questionanswering/generative/llm/Llm.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2023 Aryn
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.opensearch.searchpipelines.questionanswering.generative.llm;
+
+/**
+ * Capabilities of large language models, e.g. completion, embeddings, etc.
+ */
+public interface Llm {
+
+    ChatCompletionOutput createChatCompletion(ChatCompletionInput input);
```

Comment:
better name for function can be doChatCompletion.

createChatCompletion function is not telling what it is creating.

Reply:
The name comes straight from OpenAI:

https://platform.openai.com/docs/api-reference/chat/create.

I thought I would just follow that for familiarity, but `doChatCompletion` is fine, too.

Reply:
Thats a good argument, but doesn't mean that we need to follow what OpenAI is doing. Given that we have flexibility(as we have control) here we should name the functions properly.

Reply:
I'm fine with `doChatCompletion`.